packages/chatbot-server-mongodb-public/src/processors/generateResponseWithTools.ts

```diff
@@ -85,7 +88,7 @@ export function makeGenerateResponseWithSearchTool({
     }
     const userMessage: UserMessage = {
       role: "user",
-      content: latestMessageText,
+      content: formatUserMessageForGeneration(latestMessageText, customData),
```

Comment:
as discussed above, i recommend validating the data that you pass as the second argument w/ zod rather than passing custom data directly. better boundries between logic + cleaner code